internal/common/foraging/deerhunt.go

```diff
@@ -49,3 +57,29 @@ func deerReturn(params deerHuntParams) shared.Resources {
 	D := distuv.Bernoulli{P: params.p}        // Bernoulli RV where `P` = P(X=1)
 	return shared.Resources(D.Rand() * (1 + W.Rand()))
 }
+
+// getPopulationLinkedProbability returns the Bernoulli probability of catching a deer given the current running deer population.
+// The dynamics and variables implemented in this function are documented in the README in this package.
+func (d DeerHunt) getPopulationLinkedProbability(dhConf config.DeerHuntConfig, population uint) float64 {
+	pCritical := 1.0
+	pMax := float64(dhConf.MaxDeerPopulation) / float64(dhConf.MaxDeerPerHunt)
+	thetaCritical := dhConf.ThetaCritical
+	thetaMax := dhConf.ThetaMax
+	p := float64(population) / float64(dhConf.MaxDeerPerHunt)
+	alpha := (thetaMax - thetaCritical) / (pMax - float64(pCritical))
```

Comment:
No issues, just wanted to say I'm very impressed with this. I don't know how you came up with it all.